### PR TITLE
table improvements

### PR DESCRIPTION
**Pull request by ChristianKienle**

### CODE REVIEW

File: src/components/Table/Components/RowSelectionIndicator.tsx [hunk 1]
```diff
@@ -1,11 +1,19 @@
 import { Event, Prop, Component, Base } from '@/core';
 import { Checkbox } from '@/components/Form';
```
Comment (DamianLion) [inline src/components/Table/Components/RowSelectionIndicator.tsx]: mixed usage of `@/` annotation and relative path

Comment (ChristianKienle) [reply src/components/Table/Components/RowSelectionIndicator.tsx]: fixed

**Comment by ChristianKienle** [comment]
@DamianLion can you please review again? I made the following breaking changes:

- the items-array no longer needs an id.
- the header is configured by passing an array to the table. Either with plain strings (header label) or with a config object. The array can also be mixed.

Tests and documentation changed accordingly.

I did this change because the other technique made it impossible to easily change the displayed columns after the first render pass because `$slots` is not reactive.

**Comment by ChristianKienle** [comment]
@GertSchneider you are reviewer just so that you stay informed. No need to do anything despite enjoying the progress and warming up. :)

File: src/docs/components/ComponentReference/SlotsReference.tsx [hunk 2]
```diff
@@ -34,20 +35,22 @@ export class SlotsReference extends TsxComponent<Props> {
...
-              <span>{item.name}</span>
-            }
-          </TableCell>
-          <TableCell>{item.description}</TableCell>
-        </TableRow>
+        <row-template slot='row'>
+          <TableRow>
+            <TableCell>
+              {item.name === '' ?
+                <span style='color: rgb(200, 200, 200);'>default</span> :
```
Comment (DamianLion) [inline src/docs/components/ComponentReference/SlotsReference.tsx]: color css assigment without a class. Hard to override

Comment (ChristianKienle) [reply src/docs/components/ComponentReference/SlotsReference.tsx]: This is part of the documentation and not a component intended to be used by anyone except us.

**Pull request merged**